File: backend/internal/routes/route_model.go
package routes

import "time"

type Route struct {
	ID            string    `json:"id"`
	StartLocation string    `json:"start_location"`
	EndLocation   string    `json:"end_location"`
	Distance      string    `json:"distance"`
	CreatedAt     time.Time `json:"created_at"`
}

type CreateRouteRequest struct {
	StartLocation string `json:"start_location"`
	EndLocation   string `json:"end_location"`
	Distance      string `json:"distance"`
}

File: frontend/lib/models/route_model.dart
class RouteModel {
  final String id;

  final String startLocation;

  final String endLocation;

  final String distance;

  final String createdAt;

  RouteModel({
    required this.id,

    required this.startLocation,

    required this.endLocation,

    required this.distance,

    required this.createdAt,
  });

  factory RouteModel.fromJson(Map<String, dynamic> json) {
    return RouteModel(
      id: json["id"],

      startLocation: json["start_location"],

      endLocation: json["end_location"],

      distance: json["distance"],

      createdAt: json["created_at"],
    );
  }
}
